provisioning/transport/mqtt/devdoc/mqtt_requirements.md

```diff
@@ -0,0 +1,127 @@
+
+**SRS_NODE_PROVISIONING_MQTT_18_021: [** `queryOperationStatus` shall wait for a response for `timeoutInterval` milliseconds.  After that shall be considered a timeout. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_022: [** When `queryOperationStatus` times out, it shall disconnect the transport. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_023: [** When `queryOperationStatus` times out, it shall call callback passing in a ServiceUnavailableError. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_024: [** When waiting for responses, `queryOperationStatus` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_025: [** When waiting for responses, `queryOperationStatus` shall ignore any messages that don't match the required topic format. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_026: [** If `queryOperationStatus` receives a response with status >= 300, it shall consider the query failed and create an error using `translateError`. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_027: [** When `queryOperationStatus` receives a successful response from the service, it shall call callback passing in null and the response. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_028: [** When `queryOperationStatus` receives an error from the service, it shall disconnect the transport. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_029: [** When `queryOperationStatus` receives an error from the service, it shall call `callback` passing in the error. **]**
+
+
+## cancel(callback: (err?: Error) => void): void
+
+**SRS_NODE_PROVISIONING_MQTT_18_030: [** If `cancel` is called while the transport is disconnected, nothing will be done. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_031: [** If `cancel` is called while the transport is in the process of connecting, it shall wait until the connection is complete and then disconnect the transport. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_032: [** If `cancel` is called while the transport is connected and idle, it will disconnect the transport. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_033: [** If `cancel` is called while the transport is in the middle of a `registrationRequest` operation, it will disconnect the transport and indirectly cause `registrationRequest` call it's `callback` passing an error. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_034: [** If `cancel` is called while the transport is in the middle of a `queryOperationStatus` operation, it will disconnect the transport and indirectly cause `registrationRequest` call it's `callback` passing an error. **]**
+
+
+## transport connection
+These requirements apply whenever the transport connection is established
+
+**SRS_NODE_PROVISIONING_MQTT_18_035: [** When connecting, `Mqtt` shall set `clientId` in the `TransportConfig` object to the `registrationId`. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_036: [** When connecting, `Mqtt` shall set the `clean` flag in the base `TransportConfig` object to true. **]**
```

Comment:
>clean [](start = 80, length = 5)

i'd advocate to call this cleanSession since this is what is mostly used in the mqtt world (even though it's not the case with the mqttjs library)

Reply:
Good idea, but this is a name from mqtt common and I don't want to touch that (would it be a breaking change?)

---
In reply to: [156259325](https://github.com/Azure/azure-iot-sdk-node/pull/178#discussion_r156259325) [](ancestors = 156259325)